src/utils/jss/renderInShadowRoot.tsx

```diff
@@ -1,55 +1,119 @@
-import { create } from 'jss'
-import { createGenerateClassName, jssPreset, StylesProvider, ThemeProvider } from '@material-ui/styles'
+import { create, Jss, SheetsRegistry } from 'jss'
+import { jssPreset, StylesProvider, ThemeProvider } from '@material-ui/styles'
 import ReactDOM from 'react-dom'
 import React from 'react'
-import ConstructableStyleSheetsRenderer, {
-    applyAdoptedStyleSheets,
-    livingShadowRoots,
-} from './ConstructableStyleSheetsRenderer'
 import { getActivatedUI } from '../../social-network/ui'
 import { useValueRef } from '../hooks/useValueRef'
 import { languageSettings } from '../../components/shared-settings/settings'
 import { I18nextProvider } from 'react-i18next'
 import i18nNextInstance from '../i18n-next'
+import { portalShadowRoot } from './ShadowRootPortal'
 
-const jss = create({
-    ...jssPreset(),
-    Renderer: globalThis.location.hostname === 'localhost' ? undefined : (ConstructableStyleSheetsRenderer as any),
-})
 /**
  * Render the Node in the ShadowRoot
  * @param node React Node
  * @param shadow ShadowRoot that want to inject to
  */
 export function renderInShadowRoot(node: React.ReactNode, shadow: ShadowRoot) {
     if (shadow.mode === 'open') console.warn('Do not render with open ShadowRoot!')
-    ReactDOM.render(<RenderInShadowRootWrapper children={node} />, shadow as any)
-    livingShadowRoots.add(shadow)
-    applyAdoptedStyleSheets()
-    return () => {
-        shadow.adoptedStyleSheets = []
-        ReactDOM.unmountComponentAtNode(shadow as any)
-        livingShadowRoots.delete(shadow)
+
+    // ? how to pass props to class component?
+    // @ts-ignore
+    ReactDOM.render(<RenderInShadowRootWrapper children={node} shadow={shadow} />, shadow as any)
+    return () => ReactDOM.unmountComponentAtNode(shadow as any)
+}
+
+// ! let jss tell us if it has made an update
+class InformativeSheetsRegistry extends SheetsRegistry {
+    private callback: Set<() => void>
```

Comment:
`private callback = new Set<(() => void)>`
Then you can get rid of the whole constructor block